closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -472,6 +472,7 @@ private static void setProperties(Properties props) {
         propsMapping.put("jdk.tls.legacyAlgorithms", restricts.jdkTlsLegacyAlgorithms);
         propsMapping.put("jdk.certpath.disabledAlgorithms", restricts.jdkCertpathDisabledAlgorithms);
         propsMapping.put("jdk.security.legacyAlgorithm", restricts.jdkSecurityLegacyAlgorithm);
+        System.setProperty("com.ibm.fips.mode", restricts.jdkFipsMode);
```

Comment:
What if a user also says `-Dcom.ibm.fips.mode=140-n`? It seems wrong to ignore it if it's incompatible with the value specified in `semeru.customprofile`.

Reply:
Yes it is true that this value will override any existing values for `com.ibm.fips.mode` when a restricted security mode profile is loaded. Given that this property will be based entirely upon the profile being set I am unsure what other behavior to recommend here. Perhaps we should add a trace to this section to indicate that a user defined value was overwritten?

Reply:
I suggest use of `printStackTraceAndExit()` is appropriate. In my view, this is like asking to enable FIPS on a platform where it's not supported. Other code may have already inspected `com.ibm.fips.mode` and made decisions based on the value seen then; I don't think it's reasonable to just overwrite the property. Perhaps we want something like this:
```java
        String fipsMode = System.getProperty("com.ibm.fips.mode");
        if (fipsMode == null) {
            System.setProperty("com.ibm.fips.mode", restricts.jdkFipsMode);
        } else if (!fipsMode.equals(restricts.jdkFipsMode)) {
            printStackTraceAndExit("com.ibm.fips.mode incompatible with semeru.customprofile");
        }
```

Reply:
Yes agreed that we can make this fatal. 

I updated the code as suggested, only I changed the error message to `Property com.ibm.fips.mode is incompatible with semeru.customprofile and semeru.fips properties.` This property should not be set in the case where a user is using either default ( just `-Dsemeru.fips=true` ) or non default ( `-Dsemeru.fips=true` and `-Dsemeru.customprofile` ) properties and the restricted security mode.